secure/secure.sh

```diff
@@ -470,18 +492,50 @@ function config_mapred_site() {
   fi
 
   set_property_mapred_site 'mapreduce.jobhistory.http.policy' 'HTTPS_ONLY'
-  set_property_mapred_site 'mapreduce.ssl.enabled' 'true'
   set_property_mapred_site 'mapreduce.shuffle.ssl.enabled' 'true'
   set_property_mapred_site 'mapreduce.jobhistory.address' "${MASTER_FQDN}:10020"
   set_property_mapred_site 'mapreduce.jobhistory.webapp.http.address' "${MASTER_FQDN}:19888"
   set_property_mapred_site 'mapreduce.jobhistory.webapp.https.address' "${MASTER_FQDN}:19889"
 }
 
-function copy_keystore_files() {
-  echo "Copying keystore and truststore files from GCS."
+function copy_or_create_keystore_files() {
   mkdir "${HADOOP_CONF_DIR}"/ssl
-  gsutil cp "${keystore_uri}" "${HADOOP_CONF_DIR}"/ssl/keystore.jks
-  gsutil cp "${truststore_uri}" "${HADOOP_CONF_DIR}"/ssl/truststore.jks
+  if [[ -n "${keystore_uri}" ]]; then
+    echo "Copying keystore and truststore files from GCS."
+    gsutil cp "${keystore_uri}" "${HADOOP_CONF_DIR}"/ssl/keystore.jks
+    gsutil cp "${truststore_uri}" "${HADOOP_CONF_DIR}"/ssl/truststore.jks
+  else
+    local keystore_location="${staging_meta_info_gcs_prefix}/keystores"
+    if [[ "${FQDN}" == "${MASTER_FQDN}" ]]; then
+      echo "Generating self-signed certificate from ${FQDN}"
+      # The validity of the cert is 1800 days.
+      keytool -genkeypair -keystore keystore.jks -keyalg RSA -keysize 2048 \
+        -storepass "${keystore_password}" -keypass "${keystore_password}" -validity 1800 -alias dataproc-cert \
+        -dname "CN=*.${DOMAIN}, OU=Google Cloud Platform, O=Google, L=Mountain View, S=CA, C=US"
```

Comment:
Let's avoid using Google, etc. If these are required fields, let's use fake values.

Reply:
removed Google, GCP, etc. Only kept the wildcard CNAME.